### ISSUE TITLE
Tree examples accessibility issues

### ISSUE THREAD

**Issue by tinayuangao** [angular/components#11328]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Screen reader users could access the tree with all the information

#### What is the current behavior?
For nested tree & flat tree example:
- announced "table with 0 element" when focusing on the tree
- leaf nodes are invisible to screen-reader users

For checkbox example:
- focusing on the toggle announced "toggle undefined"

#### What are the steps to reproduce?
https://material.angular.io/components/tree/examples

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
6.0

#### Is there anything else we should know?

**Comment by mfairchild365** [comment]
Adding more details:

## Using the following screen readers
   - NVDA 2018.4.1 + Firefox 64
   - VoiceOver for IOS (12.1.3) + Safari

## Reveal the following issues:

* Buttons to add items in the "Tree with checkboxes" example do not have accessible names (all examples). All buttons must have an accessible name that describes the purpose of the button.
* Almost all information is invisible to screen-reader users (contents of the trees are not announced and are not operable)


## Steps to reproduce
1. Using either of the following screen readers
   - NVDA 2018.4.1 + Firefox 64
   - VoiceOver for IOS (12.1.3) + Safari
2. Open the [tree example](https://material.angular.io/components/tree/examples)
3. Navigate to the first tree node in the 'tree with dynamic data' example.
   - In NVDA you can press the tab key until you reach the node
   - In VoiceOver for iOS you can swipe right until you reach the node
4. Observe that NVDA announces "Treeview" and does not announce any other information about the node.
5. Observe that VoiceOVer announces "toggle undefined, button" and does not announce any treview related semantics.

## Additionally
* [Axe](https://www.deque.com/axe/) can be used to reveal further issues.
* The [ARIA Authoring Practices document](https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView) has very helpful information on expected keyboard interactions and roles, states, and properties.

**Comment by nico151999** [comment]
I was reported the same by a screen reader user. Is there some progress on this issue? I am not familiar with screen readers but this seems to make my application unusable to people using one. Unfortunately, I do not know how to debug such issues. Would be great if there will be a fix or at least a workaround. Any suggestion appreciated. Thank you!

**Comment by behofer** [comment]
Being a screenreader user myself, I have the same behaviour with NVDA in browse mode (Version 2020.3) and JAWS (Version 2020) together with Firefox and Chrome on Windows. Seems like a general issue with ARIA trees. Advice how to fix this / or for a workaround would be highly appreciated.

**Comment by andrewseguin** [comment]
The tree accessibility has been revamped to resolve existing issues. Please open new issues if there any problems are found
